docs_src/design/adr/0019-Service-List.md

```diff
@@ -0,0 +1,87 @@
+# EdgeX Service List
+
+## Status
+**Draft as of April 25, 2021**
+Proposed for Jakarta or later release.
+
+## Context
+
+Several EdgeX Foundry services today require a list of the services that are part of an EdgeX deployment.  That is to say, that several EdgeX services need to know the names and locations of other EdgeX services that are or will be part of the EdgeX.
+
+The current list of needs include:
+
+- Security bootstrapping service requires a list of services in order to know which services need a file token for Vault access and to add shared secrets to Vault.
+- System management agent service requires a list of services in order to know which services it can provide status, configuration and metrics for client requestors.  The SMA also needs to know which services to include in any start/stop/restart operations.
+- The user interface needs to know which services, specifically device and application services, are available in order to provide interaction with those services, and potentially to display some metrics on operational services.
+- The CLI needs to know which services are operational (for some interaction) and which device services and application services to include in start/stop operations.
+
```

Comment:
core-metadata needs (and currently maintains) a list of device-type services. This is used to validate device additions (the device definition must include a valid named device service to which it is associated). It is also needed when routing device commands to the appropriate service.

(Should the existing facility in core-metadata be expanded to cover registration of all services?)

Reply:
good call out Iain. Added this consideration.